### PR TITLE
Fixes #35313 - CV Version Compare - add other content types

### PR DESCRIPTION
**Pull request by tazhibaevaaliya**
#### What are the changes introduced in this pull request?
Extended CV Version comparison feature to other content types, like Ansible collections and Python packages

#### What are the testing steps for this pull request?
1. Publish two versions so that either of them has all the 9 content types (RPM Packages, RPM Package groups, Files, Docker tags, Debs, Errata, Module streams, Python packages and Ansible collections). 
2. Select those two versions in the list of all versions 
3. Hit "Compare" button
4. Ensure that you are directed to the table with all 9 tabs for all the content types
5. Try switching the tabs and ensure that all of the tabs render required information

### CODE REVIEW

**Comment by jeremylenz** [comment]
@tazhibaevaaliya Looks like this needs a rebase now that the other PR is merged..

**Comment by tazhibaevaaliya** [comment]
@jeremylenz @sjha4  rebased it. Ready for the review :sweat_smile:

**Comment by jeremylenz** [comment]
@tazhibaevaaliya can you point me to the URLs I can sync to get all of the content types? :pray:

**Comment by tazhibaevaaliya** [comment]
Sure. Here is the list of repos I have: 
repo1: https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/
repo2: https://quay.io/
repo3: https://fixtures.pulpproject.org/debian/
repo4: https://fixtures.pulpproject.org/file/
repo5: https://galaxy.ansible.com/  AND requirements: 
---
collections:
- marmorag.ansodium

repo6: https://pypi.org AND includes: django~=2.0
repo7: https://partha.fedorapeople.org/test-repos/pteradactyl/

Hope this helps!

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js [hunk 1]
```diff
@@ -169,6 +170,18 @@ export const getFilesComparison = (versionOne, versionTwo, params) => {
...
 };
 
+export const getContentComparison = (pluralLabel, versionOne, versionTwo, params) => {
+  const versions = { content_view_version_ids: [versionOne, versionTwo] };
+  const apiParams = { ...versions, ...params };
+  const apiUrl = `/${pluralLabel}/compare`;
+  return get({
+    key: genericContentCompareKey(pluralLabel, versionOne, versionTwo),
+    params: apiParams,
+    errorToast: error => __(`Something went wrong while retrieving the packages! ${getResponseErrorMsgs(error.response)}`),
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:180]:
```suggestion
    errorToast: error => __(`Something went wrong while retrieving the content! ${getResponseErrorMsgs(error.response)}`),
```

**Comment by sjha4** [comment]
[test katello]

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js [hunk 2]
```diff
@@ -169,6 +170,18 @@ export const getFilesComparison = (versionOne, versionTwo, params) => {
   });
 };
 
+export const getContentComparison = (pluralLabel, versionOne, versionTwo, params) => {
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:173]:
```suggestion
export const getGenericContentComparison = (pluralLabel, versionOne, versionTwo, params) => {
```

File: webpack/scenes/ContentViews/Details/Versions/Compare/__tests__/CVVersionCompare.test.js [hunk 3]
```diff
@@ -368,31 +399,31 @@ test('Can select two versions with no content and click compare button', async (
   const scope = nockInstance
     .get(cvVersions)
     .query(true)
-    .reply(200, emptyStateCVVersionsData);
+    .reply(200, cvVersionsData);
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/__tests__/CVVersionCompare.test.js:402]: Why are we doing this? The 'Can select two versions with no content and click compare button' test name doesn't reflect what the test is doing anymore?

Comment (sjha4) [reply webpack/scenes/ContentViews/Details/Versions/Compare/__tests__/CVVersionCompare.test.js:402]: Oh..nevermind..I see version 1 and 2 in fixture are empty so this should be good.

**Comment by sjha4** [comment]
One small nitpick..Works well..Ready to go with one name change.

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js (hunk 2 above)
Comment (sjha4) [reply webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:173]: @tazhibaevaaliya : Just add this to your other PR and we can merge this so you don't can stop worrying about rebasing and stuff..

**Review by sjha4 — APPROVED** [review]
Nice work @tazhibaevaaliya ! Merging this. 🎉

**Pull request merged**